torch/csrc/jit/runtime/symbolic_shape_registry.cpp

```diff
@@ -49,6 +49,43 @@ const std::string shape_compute_functions =
             out.append(elem)
           return out
 
+        def unary_one_unused_input(self: List[int], inp0: Any):
+          out: List[int] = []
+          for elem in self:
+            out.append(elem)
+          return out
+
+        def unary(self: List[int]):
+          out: List[int] = []
+          for elem in self:
+            out.append(elem)
+          return out
+
+        def view(self: List[int], sizes: List[int]):
+          # TODO: add assertions to check whether requested dims are valid
+          out: List[int] = []
+          for elem in sizes:
+            if elem == -1:
+              # TODO: support -1 in view dimensions
+              raise AssertionError("Shape function doesn't support -1 view dims yet")
+            out.append(elem)
+          return out
+
+        def mean_dim(self: List[int], dims: List[int], keep_dim: bool, dt : Any):
```

Comment:
where is this code taken from ? it doesn't look that similar to the logic i find in eager: https://codebrowser.bddppq.com/pytorch/pytorch/aten/src/ATen/native/ReduceOpsUtils.h.html#_ZN2at6native23create_reduction_resultERKNS_6TensorEN3c108ArrayRefIlEEbNS4_10ScalarTypeE

Reply:
I wrote it from scratch, do you think it's different from this one?
https://github.com/pytorch/pytorch/blob/1798ff02e430ad4c9ca1aff218913142da39d4a8/aten/src/ATen/native/ReduceOpsUtils.h#L137-L145
I didn't explicitly create the mask like the code for the eager mode, but i'm not sure it's different.

Reply:
it might be ... but we can take it as a follow up to replace this with the implementation logic, as with `view`. it's much easier to reason about when there is a direct 1-1 correspondence... 

future PRs we can be more strict. adding OpInfo tests (which i'm working on) should also help flush out partially-covered semantics

Reply:
upon further comparison LGTM...